lib/resty/healthcheck.lua

```diff
@@ -433,6 +441,28 @@ function checker:clear()
 end
 
 
+function checker:delayed_clear(delay)
+  assert(tonumber(delay), "no delay provided")
+
+  return locking_target_list(self, function(target_list)
+    local purge_time = ngx_now() + delay
+
+    -- add purge time to all targets
+    for _, target in ipairs(target_list) do
+      target.purge_time = purge_time
```

Comment:
What would happen if `target.purge_time` is already set? Is it even possible to encounter that condition here?

If it _is_ possible, then it seems like `checker:delayed_clear()` would postpone the target's purge time, which may or may not be our intent.

Reply:
That makes good sense to me.

I was somewhat worried that this might make it possible to postpone target purging indefinitely, leaving the target in limbo by constantly resetting its `purge_time` if config changes from Kong happen before the target has a chance to expire.

The way you've explained it here makes that sound like a non-issue (the desired behavior, even), because calling `checker:delayed_clear()` more than once in short succession probably implies that the checker has been removed/re-added several times, and in that case we probably _do_ want to continue to postpone the target expiration until the config churn settles down.